src/react/account/AccountUpdateUser.tsx

```diff
@@ -0,0 +1,191 @@
+// @flow
+import FormContainer, * as F from '../ui-elements/FormLayout';
+import React, { useRef } from 'react';
+import { clearError, handleErrorMessage, networkEnd } from '../actions';
+import { useDispatch, useSelector } from 'react-redux';
+import type { AppState } from '../../types/state';
+import { Banner } from '@scality/core-ui';
+import { Button } from '@scality/core-ui/dist/next';
+import Joi from '@hapi/joi';
+import { joiResolver } from '@hookform/resolvers';
+import { useForm } from 'react-hook-form';
+import { useOutsideClick } from '../utils/hooks';
+import { useIAMClient } from '../IAMProvider';
+import { queryClient } from '../App';
+import { useMutation } from 'react-query';
+import { useHistory, useParams } from 'react-router-dom';
+import { notFalsyTypeGuard } from '../../types/typeGuards';
+const regexpName = /^[\w+=,.@ -]+$/;
+const schema = Joi.object({
+  name: Joi.string()
+    .label('Name')
+    .required()
+    .min(2)
+    .max(64)
+    .regex(regexpName)
+    .message('Invalid Name'),
+});
+
+const AccountUpdateUser = () => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const IAMClient = useIAMClient();
+  const { IAMUserName, accountName } = useParams<{ IAMUserName: string, accountName: string }>();
+  const { register, handleSubmit, errors } = useForm({
+    resolver: joiResolver(schema)
+  });
+  const updateUserMutation = useMutation(
+    (newUserName: string) => {
+      const oldUserName = IAMUserName;
+      return notFalsyTypeGuard(IAMClient).updateUser(newUserName, oldUserName)
+        .then(() => {
```

Comment:
Right now when you arrive directly on the page and try to edit the name it will crash.
In order to handle this "edge cases" you should check if you have data first.

In the `then`, you should first check : 
```suggestion
        .then(() => {
          const olddata = queryClient.getQueryData([
            'listIAMUsers',
            accountName,
          ]);
          if (olddata) {
             queryClient.setQueryData...
             [...]
          }
```